src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -1650,12 +1651,35 @@ int32_t SystemNative_Accept(intptr_t socket, uint8_t* socketAddress, int32_t* so
     while ((accepted = accept4(fd, (struct sockaddr*)socketAddress, &addrLen, SOCK_CLOEXEC)) < 0 && errno == EINTR);
 #else
     while ((accepted = accept(fd, (struct sockaddr*)socketAddress, &addrLen)) < 0 && errno == EINTR);
+#if defined(FD_CLOEXEC)
+    // macOS does not have accept4 but it can set _CLOEXEC on descriptor.
```

Comment:
Maybe add a note that this is _best effort_. An exec between accept and fcntl will still inherit the socket.

Reply:
good point since the calls are not atomic unlike accept4. Should we than simply ignore failures?

Reply:
Or `assert` the return value is `0` since the call should never fail.